apps/omg/lib/signature.ex

```diff
@@ -0,0 +1,124 @@
+# Copyright 2019 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.Signature do
```

Comment:
Hm, isn't there any other external source to get this code from, this looks pretty low-level? I'm assuming the reason is that `exthereum/blockchain` got discontinued, or?

If there isn't, maybe we should consider starting one?

Lastly, we had all the signing functions grouped in `support` dirs, making them unavailable in `:prod`, to mark their non-production. With the inclusion of this module, we're slightly loosening this prod vs dev boundary. 

If we had an external repo hold this wrapper `OMG.Signature`, which then get's adapted in `Crypto` (recovery) and `DevCrypto` (signing, priv key handling) modules on our side, thing could have been more transparent.

Reply:
I don't think it makes much sense, to have an external source for three functions, that are basically libsecp256k1 wrappers. they're not going to change.
`exthereum/blockchain` was discontinued,because they're part of mana umbrella applications - basically for the same reason. it had little sense supporting them separately. 

I think we can get the same approach if we just keep `recover_public/4` in the Signature. every other function is used in test only and thats where it should be moved to.